CookieMediator/src/main/java/org/wso2/carbon/connector/creatiocrm/headers/BPMCSRFCookie.java

```diff
@@ -0,0 +1,128 @@
+
+
+/*
+ * Copyright (c) 2022, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.wso2.carbon.connector.creatiocrm.headers;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.apache.synapse.core.axis2.Axis2MessageContext;
+import org.apache.synapse.mediators.AbstractMediator;
+import org.apache.synapse.MessageContext;
+
+/**
+ * Class BPMCSRFCookie which to derive the HTTP headers from the Creatio Authentication response
+ * Multiple Set-cookie HTTP headers contains Authentication information for rest of the API calls and process those and assign
+ * to relevant context value in order to use in the Creatio connector implementation
+ */
+public class BPMCSRFCookie extends AbstractMediator {
+
+    private static final Log log = LogFactory.getLog(BPMCSRFCookie.class);
+    private static String COOKIE_NAME = "set-cookie";
+    private static String COOKIE_NAME_CAMELCASE = "Set-Cookie";
+    private static String COOKIE_BPMCSRF = "BPMCSRF=";
+    private static String AUTH_COMPLETE_PROPERTY = "authCompleteCookie";
+    private static String BPMSESSIONID = "BPMSESSIONID";
+    private static String USERNAME = "UserName";
+    private static String BPMLOADER = "BPMLOADER";
+    private static String ASPXAUTH = ".ASPXAUTH";
+    private static String BPMCSRF = "BPMCSRF";
```

Comment:
Since these are constant, Shall we add the final keyword as well?

Reply:
Added the final keyword